### PR TITLE
Refactor code Pull request

### PR DESCRIPTION
**Pull request by HERMON-1995**
In this project, we used newly acquired knowledge and refactor our previous code. The following are the changes:

- Followed YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Followed SOLID principles.
- Payed special attention to the "Single-responsibility Principle".

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -45,38 +48,39 @@ def create_person
     end
   end
 
+  def select_person_type
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    gets.chomp.to_i
+  end
```
Comment (Ibizugbe) [inline app.rb]: Nice job so far Team👍. You have followed the Single responsibility rule so far but have deviated a bit in this file😉. Don't forget we learned from the lesson on [best practices and solid](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) that when respecting the Single responsibility rule, every class should have just one responsibility😉. currently, you have assigned the responsibility of collecting data to the `action` class and input validation to the `main` class😉, but here you still have the same responsibility assigned in the `app`🥲. Kindly ensure everything that has to do with receiving input is done in the `action` class and validating input is done in the `main` class👍.

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @HERMON-1995 👋👋👋, 

tryagain


Good job so far!👍
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍.
- [x] Git flow have been used.👍.
- [x] App still works fine👍.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (HERMON-1995) [reply app.rb]: Thank you for your insightful review! The code has been refactored accordingly. 😄

**Review by levy002 — APPROVED** [review]
## Hello @HERMON-1995 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

  unnamed


## Highlights
- Implemented suggested changes :heavy_check_mark:
- No linter error :heavy_check_mark: 
 
## Status: APPROVED :fire:  :fire: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HERMON-1995** [comment]
Thank you for approving our project!😄

**Pull request merged**